code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -352,18 +372,55 @@ All foods are distributed among various categories. Use common sense.
 		if(contents.len >= 20)
 			to_chat(user, "<span class='warning'>[src] is full.</span>")
 			return 0
+		user.visible_message("<span class='notice'>[user.name] begins inserting [W] into [src].</span>", \
+						"<span class='notice'>You start to insert the [W] into \the [src].</span>")
+		if(!do_after(user, 1.5 SECONDS, target = src))
+			return 0			
 		to_chat(user, "<span class='notice'>You slip [W] inside [src].</span>")
 		user.transferItemToLoc(W, src)
+		log_message("[key_name(user)] inserted [W.name] into [src.name] at [AREACOORD(src)]", LOG_ATTACK)
 		add_fingerprint(user)
 		contents += W
-		stored_item = 1
+		stored_item = TRUE
 		return 1 // no afterattack here
 
+/obj/item/reagent_containers/food/snacks/store/attack(mob/living/carbon/M, mob/living/carbon/user, def_zone)
+	if(!..())
+		return
+	/// What are the odds we eat glass? - [Bitecount / Max number of bites] * 100
+	var/bad_chance_of_discovery = (bitecount / (volume_on_creation / bitesize))*100	//the closer you get to finishing it, the higher the chance you bite into it
+	/// What are the odds we see the glass but don't bite it? - ([Bitecount / Max number of bites] * 100) - 50
+	var/good_chance_of_discovery = bad_chance_of_discovery - 50	//the closer you get to finishing it, the more likely you can see what is in it
+	/// We've found the item, and plan on remove it
+	var/discovered = FALSE
+
+	if(stored_item)
+		for(var/obj/item/I in contents)
+			if(istype(I, /obj/item/reagent_containers/food/snacks))
+				return 0
+			if(prob(good_chance_of_discovery))
+				discovered = TRUE
+				to_chat(M, "<span class='warning'>It feels like there's something in this [src.name]...!</span>")
+
+			else if(prob(bad_chance_of_discovery))
+				//sharp things, nuke disks, pills, syringes, anything with mats, tiny things, bullets, organs, drink cans, bottles
+				log_message("[key_name(user)] just fed [key_name(M)] a/an [I.name] which was hidden in [src.name] at [AREACOORD(src)]", LOG_ATTACK)
+				discovered = I.on_accidental_consumption(M, user, src)
+
+			if(!QDELETED(I) && discovered)
+				contents -= I
+				stored_item = FALSE
```

Comment:
what if there are still other items in the cake?

Reply:
in it's current implementation it doesn't actually let you store multiple items in a food container (if checks if stored_item is true and doesn't allow you to insert if it is)

Reply:
still, just as a sanity check...